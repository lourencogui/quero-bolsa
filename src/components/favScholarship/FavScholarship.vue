<template>
  <div>
    <div class="container">
      <div class="breadcrumb">
        <div class="breadcrumb__item">
          <a class="breadcrumb__text text--s text--secondary text--bold">Home</a>
        </div>
        <div class="breadcrumb__item">
          <a class="breadcrumb__text text--s text--secondary text--bold">Minha conta</a>
        </div>
        <div class="breadcrumb__item breadcrumb__item--selected">
          <a class="breadcrumb__text text--s">Bolsas favoritas</a>
        </div>
      </div>

      <div class="page-info">
        <h2>Bolsas Favoritas</h2>
        <p>Adicione bolsas de cursos e faculdades do seu interesse e receba atualizações com as melhores ofertas disponíveis</p>
      </div>
      <div class="filter">
        <div class="filter__item filter__item--selected">
          <p>Todos os semetres</p>
        </div>
        <div class="filter__item">
          <p>2º semetre de 2019</p>
        </div>
        <div class="filter__item">
          <p>1º semetre de 2019</p>
        </div>
      </div>

      <div class="scholarships">
        <div class="scholarships__new" @click="openNewScholarship">
          <div>
            <font-awesome-icon icon="plus-circle" color="#18acc4" size="3x"/>
          </div>
          <div class="scholarships__new__info">
            <h2>Adicionar bolsa</h2>
            <strong>Clique para adicionar bolsas de cursos do seu interesse</strong>
          </div>
        </div>
        <FavScholarshipItem/>
      </div>
    </div>
    <NewScholarship ref="newScholarship"/>
  </div>
</template>
<style lang="scss" scoped>
@import "./_favScholarship.scss";
</style>


<script>
import FavScholarshipItem from "./components/FavScholarshipItem/FavScholarshipItem.vue";
import NewScholarship from "../newScholarship/NewScholarship.vue";

export default {
  name: "favScholarship",
  components: {
    FavScholarshipItem,
    NewScholarship
  },
  methods: {
    // addNew() {},
    openNewScholarship() {
      this.$refs.newScholarship.changeVisibility();
    }
  }
};
</script>
