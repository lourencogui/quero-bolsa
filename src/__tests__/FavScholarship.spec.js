import { shallowMount } from "@vue/test-utils";
import FavScholarship from "../components/favScholarship/FavScholarship";
import { FontAwesomeIcon } from "@fortawesome/vue-fontawesome";

describe("FavScholarship", () => {
  it("component render properly", () => {
    const wrapper = shallowMount(FavScholarship, {
      stubs: {
        FontAwesomeIcon
      }
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("function removeItem should remove one item from favorite array", () => {
    const wrapper = shallowMount(FavScholarship, {
      stubs: {
        FontAwesomeIcon
      }
    });
    const favoriteArr = [
      {
        id: 1,
        full_price: 2139.64,
        price_with_discount: 706.08,
        discount_percentage: 67.0,
        start_date: "01/08/2019",
        enrollment_semester: "2019.2",
        enabled: true,
        course: {
          name: "Engenharia Mecânica",
          kind: "Presencial",
          level: "Bacharelado",
          shift: "Noite"
        },
        university: {
          name: "UNIP",
          score: 4.5,
          logo_url: "https://www.tryimg.com/u/2019/04/16/unip.png"
        },
        campus: {
          name: "Jardim das Indústrias",
          city: "São José dos Campos"
        }
      },
      {
        id: 2,
        full_price: 2139.64,
        price_with_discount: 706.08,
        discount_percentage: 67.0,
        start_date: "01/08/2019",
        enrollment_semester: "2019.2",
        enabled: true,
        course: {
          name: "Engenharia Mecânica",
          kind: "Presencial",
          level: "Bacharelado",
          shift: "Noite"
        },
        university: {
          name: "UNIP",
          score: 4.5,
          logo_url: "https://www.tryimg.com/u/2019/04/16/unip.png"
        },
        campus: {
          name: "Jardim das Indústrias",
          city: "São José dos Campos"
        }
      }
    ];
    wrapper.setData({
      favoriteScholarships: favoriteArr,
      filteredFavoriteScholarships: favoriteArr
    });
    wrapper.vm.removeFavorite(1);
    expect(wrapper.vm.favoriteScholarships.length).toBe(1);
  });
});
