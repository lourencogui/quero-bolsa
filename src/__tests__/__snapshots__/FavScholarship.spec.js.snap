// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavScholarship component render properly 1`] = `
<div class="wrapper">
  <div class="container">
    <div class="breadcrumb">
      <div class="breadcrumb__item"><a class="breadcrumb__text text--s text--secondary text--bold">Home</a></div>
      <div class="breadcrumb__item"><a class="breadcrumb__text text--s text--secondary text--bold">Minha conta</a></div>
      <div class="breadcrumb__item breadcrumb__item--selected"><a class="breadcrumb__text text--s">Bolsas favoritas</a></div>
    </div>
    <div class="page-info">
      <h2 class="text text--xl">Bolsas Favoritas</h2>
      <p class="text text--s text--lightWeight">
        Adicione bolsas de cursos e faculdades do seu interesse e receba
        atualizações com as melhores ofertas disponíveis
      </p>
    </div>
    <div class="filter">
      <div class="filter__item filter__item--selected">
        <p class="text text--s">Todos os semetres</p>
      </div>
      <div class="filter__item">
        <p class="text text--s">2º semetre de 2019</p>
      </div>
      <div class="filter__item">
        <p class="text text--s">1º semetre de 2020</p>
      </div>
    </div>
    <div class="scholarships">
      <div class="scholarships__new">
        <div>
          <!---->
        </div>
        <div class="scholarships__new-info">
          <h2 class="text text--lg">Adicionar bolsa</h2> <strong class="text text--xs">Clique para adicionar bolsas de cursos do seu interesse</strong>
        </div>
      </div>
    </div>
  </div>
  <newscholarship-stub></newscholarship-stub>
</div>
`;
